service/src/main/java/de/adorsys/psd2/sandbox/xs2a/testdata/TestDataMapper.java

```diff
@@ -97,21 +102,36 @@ public SpiTransaction mapTransactionToSpiTransaction(
   /**
    * Maps TestData Balance List to SpiAccountBalance List.
    *
-   * @param balances TestData List of Balance
+   * @param account         account object to be mapped
+   * @param consentBalances list of balances within consent
    * @return SpiAccountBalance List
    */
-  public List<SpiAccountBalance> mapBalanceListToSpiBalanceList(List<Balance> balances) {
+  public List<SpiAccountBalance> mapBalanceListToSpiBalanceList(Account account,
+      List<SpiAccountReference> consentBalances) {
+    List<Balance> balances = account.getBalances();
     List<SpiAccountBalance> spiAccountBalances = new ArrayList<>();
-    for (Balance balance : balances) {
-      SpiAccountBalance spiBalance = new SpiAccountBalance();
-      SpiAmount spiAmount = new SpiAmount(balance.getBalanceAmount().getCurrency(),
-          balance.getBalanceAmount().getAmount());
-      spiBalance.setSpiBalanceAmount(spiAmount);
-      spiBalance.setSpiBalanceType(SpiBalanceType.valueOf(balance.getBalanceType().name()));
-      spiAccountBalances.add(spiBalance);
+    if (isBalanceAccessAllowed(account, consentBalances)) {
+      for (Balance balance : balances) {
+        SpiAccountBalance spiBalance = new SpiAccountBalance();
+        SpiAmount spiAmount = new SpiAmount(balance.getBalanceAmount().getCurrency(),
+            balance.getBalanceAmount().getAmount());
+        spiBalance.setSpiBalanceAmount(spiAmount);
+        spiBalance.setSpiBalanceType(SpiBalanceType.valueOf(balance.getBalanceType().name()));
+        spiAccountBalances.add(spiBalance);
+      }
+      return spiAccountBalances;
     }
+    return new ArrayList<>();
```

Comment:
I'd strongly recommend to do `if (!isBalanceAccessAllowed(account, consentBalances)) { return new ArrayList(); }` at beginning of the function (guard clause). Makes it way easier to follow the flow.

Reply:
Makes sense. Changed it.